zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java

```diff
@@ -55,84 +68,189 @@ private ZeebeRocksDbFactory(
-                  + "[Compiled ColumnFamilyOptions: %s; User-provided ColumnFamilyOptions: %s]. "
-                  + "See RocksDB's cf_options.h and options_helper.cc for available keys and values.",
-              columnFamilyOptionProps, userProvidedColumnFamilyOptions));
+  public ColumnFamilyOptions createColumnFamilyOptions(final List<AutoCloseable> closeables) {
+
+    final var hasUserOptions = !userProvidedColumnFamilyOptions.isEmpty();
+
+    if (hasUserOptions) {
+      final var columnFamilyOptionProps = new Properties();
+      // apply custom options
+      columnFamilyOptionProps.putAll(userProvidedColumnFamilyOptions);
+
+      final var columnFamilyOptions =
+          ColumnFamilyOptions.getColumnFamilyOptionsFromProps(columnFamilyOptionProps);
+      if (columnFamilyOptions == null) {
+        throw new IllegalStateException(
+            String.format(
+                "Expected to create column family options for RocksDB, "
+                    + "but one or many values are undefined in the context of RocksDB "
+                    + "[Compiled ColumnFamilyOptions: %s; User-provided ColumnFamilyOptions: %s]. "
+                    + "See RocksDB's cf_options.h and options_helper.cc for available keys and values.",
+                columnFamilyOptionProps, userProvidedColumnFamilyOptions));
+      }
+      return columnFamilyOptions;
     }
-    return columnFamilyOptions;
+
+    return createDefaultColumnFamilyOptions(closeables);
+  }
+
+  private ColumnFamilyOptions createDefaultColumnFamilyOptions(
+      final List<AutoCloseable> closeables) {
+    final var columnFamilyOptions = new ColumnFamilyOptions();
+
+    // given
+    final long totalMemoryBudget = 512 * 1024 * 1024L; // make this configurable
+    // recommended by RocksDB, but we could tweak it; keep in mind we're also caching the indexes
+    // and filters into the block cache, so we don't need to account for more memory there
+    final long blockCacheMemory = totalMemoryBudget / 3;
+    // flushing the memtables is done asynchronously, so there may be multiple memtables in memory,
+    // although only a single one is writable. once we have too many memtables, writes will stop.
+    // since prefix iteration is our bread n butter, we will build an additional filter for each
+    // memtable which takes a bit of memory which must be accounted for from the memtable's memory
+    final int maxConcurrentMemtableCount = 6;
+    final double memtablePrefixFilterMemory = 0.15;
+    final long memtableMemory =
+        Math.round(
+            ((totalMemoryBudget - blockCacheMemory) / (double) maxConcurrentMemtableCount)
+                * (1 - memtablePrefixFilterMemory));
+
+    final TableFormatConfig tableConfig = createTableFormatConfig(closeables, blockCacheMemory);
+
+    return columnFamilyOptions
+        // prefix seek must be fast, so allocate an extra 10% of a single memtable budget to create
```

Comment:
Oops, see it says `extra 10%`, but that comes from `memtablePrefixFilterMemory`, which is now 15% :see_no_evil: 

I guess we should comment that above? idk, but it's probably confusing atm